File: eitaa_bot.py
import requests
import time
import jdatetime  # برای تاریخ شمسی

class EitaaBot:
    def __init__(self, token):
        self.token = token
        self.api_url = 'https://eitaayar.ir/api/'

    def get_updates(self):
        url = f"{self.api_url}{self.token}/getUpdates"
        response = requests.post(url)
        return response.json()

    def send_message(self, chat_id, text):
        url = f"{self.api_url}{self.token}/sendMessage"
        data = {
            'chat_id': chat_id,
            'text': text
        }
        response = requests.post(url, data=data)
        return response.json()

    def send_file(self, chat_id, file_path, caption='', title=''):
        url = f"{self.api_url}{self.token}/sendFile"
        files = {
            'file': open(file_path, 'rb')
        }
        data = {
            'chat_id': chat_id,
            'caption': caption,
            'title': title
        }
        response = requests.post(url, files=files, data=data)
        return response.json()

    def process_message(self, message):
        chat_id = message.get('chat_id')
        text = message.get('text', '')

        if text.lower() == '/start':
            self.send_message(chat_id, 
                "سلام! به ربات ما خوش آمدید.\n\n"
                "از دستورات زیر می‌توانید استفاده کنید:\n"
                "/help - راهنما\n"
                "/time - زمان فعلی\n"
                "/about - درباره ما")

        elif text.lower() == '/help':
            self.send_message(chat_id,
                "راهنمای دستورات:\n"
                "/start - شروع\n"
                "/time - نمایش زمان\n"
                "/about - درباره ما")

        elif text.lower() == '/time':
            current_time = jdatetime.datetime.now().strftime("%Y/%m/%d %H:%M:%S")
            self.send_message(chat_id, f"زمان فعلی: {current_time}")

        elif text.lower() == '/about':
            self.send_message(chat_id, "این یک ربات نمونه است که با API ایتا ساخته شده است.")

        elif 'سلام' in text:
            self.send_message(chat_id, "سلام! چطور می‌تونم کمکتون کنم؟")

        elif 'خداحافظ' in text:
            self.send_message(chat_id, "خداحافظ! روز خوبی داشته باشید.")

        else:
            self.send_message(chat_id, "پیام شما دریافت شد. برای دیدن لیست دستورات، /help را وارد کنید.")

    def handle_updates(self):
        updates = self.get_updates()
        if 'messages' in updates:
            for message in updates['messages']:
                self.process_message(message)

def main():
    # توکن خود را اینجا قرار دهید
    token = 'bot368550:131ff932-cf46-41ec-83ca-e9f8381784c1'
    bot = EitaaBot(token)

    # تست ارسال پیام
    chat_id = input("لطفا chat_id مقصد را وارد کنید: ")
    test_message = "این یک پیام تستی است! 😊"
    
    print("در حال ارسال پیام تستی...")
    response = bot.send_message(chat_id, test_message)
    print("پاسخ سرور:", response)

    # اگر می‌خواهید ربات به کار عادی خود ادامه دهد، این قسمت را هم اضافه کنید
    print("\nربات شروع به کار کرد...")
    while True:
        try:
            bot.handle_updates()
            time.sleep(2)
        except Exception as e:
            print(f"خطا: {e}")
            time.sleep(5)

if __name__ == '__main__':
    main()

File: eitaa_bot.php
<?php

require_once 'jdate.php'; // یا هر کتابخانه تاریخ شمسی دیگر

class EitaaBot {
    private $token;
    private $api_url = 'https://eitaayar.ir/api/';

    public function __construct($token) {
        $this->token = $token;
    }

    // دریافت پیام‌های ورودی
    public function getUpdates() {
        $url = $this->api_url . $this->token . '/getUpdates';
        return $this->sendRequest($url, []);
    }

    // پردازش پیام‌های دریافتی
    public function handleUpdates() {
        $updates = $this->getUpdates();
        
        if (isset($updates['messages'])) {
            foreach ($updates['messages'] as $message) {
                $this->processMessage($message);
            }
        }
    }

    // پردازش هر پیام
    private function processMessage($message) {
        $chat_id = $message['chat_id'];
        $text = $message['text'] ?? '';

        switch (strtolower($text)) {
            case '/start':
                $this->sendMessage($chat_id, "سلام! به ربات ما خوش آمدید.\n\nاز دستورات زیر می‌توانید استفاده کنید:\n/help - راهنما\n/time - زمان فعلی\n/about - درباره ما");
                break;

            case '/help':
                $this->sendMessage($chat_id, "راهنمای دستورات:\n/start - شروع\n/time - نمایش زمان\n/about - درباره ما");
                break;

            case '/time':
                $time = jdate('Y/m/d H:i:s');
                $this->sendMessage($chat_id, "زمان فعلی: $time");
                break;

            case '/about':
                $this->sendMessage($chat_id, "این یک ربات نمونه است که با API ایتا ساخته شده است.");
                break;

            default:
                // پاسخ به پیام‌های معمولی
                if (strpos($text, 'سلام') !== false) {
                    $this->sendMessage($chat_id, "سلام! چطور می‌تونم کمکتون کنم؟");
                } elseif (strpos($text, 'خداحافظ') !== false) {
                    $this->sendMessage($chat_id, "خداحافظ! روز خوبی داشته باشید.");
                } else {
                    $this->sendMessage($chat_id, "پیام شما دریافت شد. برای دیدن لیست دستورات، /help را وارد کنید.");
                }
                break;
        }
    }

    // ارسال پیام متنی
    public function sendMessage($chat_id, $text) {
        $url = $this->api_url . $this->token . '/sendMessage';
        $data = [
            'chat_id' => $chat_id,
            'text' => $text
        ];
        return $this->sendRequest($url, $data);
    }

    // ارسال فایل
    public function sendFile($chat_id, $file_path, $caption = '', $title = '') {
        $url = $this->api_url . $this->token . '/sendFile';
        $data = [
            'chat_id' => $chat_id,
            'file' => new CURLFile(realpath($file_path)),
            'caption' => $caption,
            'title' => $title
        ];
        return $this->sendRequest($url, $data);
    }

    private function sendRequest($url, $data) {
        $ch = curl_init($url);
        curl_setopt($ch, CURLOPT_POST, true);
        curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
        
        $response = curl_exec($ch);
        curl_close($ch);
        
        return json_decode($response, true);
    }
}

// نحوه استفاده از ربات تعاملی
$token = 'bot368550:cd9b867f-e504-404c-8a23-97ab478ba77a'; // توکن خودتون رو اینجا بذارید
$bot = new EitaaBot($token);

// اجرای حلقه اصلی ربات
while (true) {
    $bot->handleUpdates();
    sleep(2); // تاخیر 2 ثانیه‌ای بین هر بررسی
} 